### ISSUE TITLE
OAuth

### ISSUE THREAD

**Issue by jasmingacic** [kubeshop/kusk-gateway#401]
Should resolve or partially resolve #166 .

**Comment by aabedraba** [comment]
OAuth on hold, trying to fix #515

**Comment by aabedraba** [comment]
@mbana to put more status on this issue

**Comment by mbana** [comment]
Information about the `grant_type`s: <https://auth0.com/docs/get-started/applications/application-grant-types>.

We want/need `client_credentials`: <https://auth0.com/docs/get-started/authentication-and-authorization-flow/client-credentials-flow>. Envoy only supports `authorization_code`: <https://github.com/envoyproxy/envoy/blob/main/source/extensions/filters/http/oauth2/oauth_client.cc#L25> - more info on the `authorization_code` `grant_type` available at: <https://auth0.com/docs/get-started/authentication-and-authorization-flow/authorization-code-flow>.

If you look at the diagrams of both the `grant_type`s you will notice `client_credentials` supports Machine-2-Machine communication.

https://techhenzy.com/envoy-oauth-filter/:

> Initially the OAuth filter will only support the Authorization Code grant type. Future iterations may implement the following grant types:
> ...
> * Client Credential

I think what we need is for this, https://github.com/envoyproxy/envoy/issues/21851, PR to get implemented. To that end, I've left a comment, <https://github.com/envoyproxy/envoy/issues/21851#issuecomment-1187682268> expressing interest in this PR.

**Comment by mbana** [comment]
OAuth2 is blocked on the folks at Envoy fixing an upstream issue with Envoy Proxy itself. Please see:

* <https://github.com/envoyproxy/envoy/issues/22678>
* <https://github.com/envoyproxy/go-control-plane/issues/581>

We can't progress further until the above have been fixed.

See this, <https://github.com/kubeshop/kusk-gateway/pull/569>, pull request for more information.

**Comment by jasmingacic** [comment]
This has been closed by https://github.com/kubeshop/kusk-gateway/pull/569
